### ISSUE TITLE
Add XGBoost inference support 

### ISSUE THREAD

**Issue by yanndupis** [tf-encrypted/moose#633]
# XGBoost Inference support
In order to support XGBoost, following this investigation https://github.com/capeprivacy/research/issues/103, here are the main features we need to support. We make the distinction between regression, binary classification and multi-class classification tasks as they differed by their output. 

## Regression
- [x] Convert an XGBoost model (ensemble of trees) saved_model or ONNX file to a tensor representation in order to later build  the model as Moose graph #636.
- [x] Express XGBoost model as a Moose computation from a tensor representation of the model. https://github.com/tf-encrypted/runtime/issues/637
-  [x] Generalize comparison operation to not only work with 0. And expose it to the EDSL. #638 
- [x] Implement a multiplexer operation to select appropriate node's child #694
- [x] Expose replicated `index_axis` operation to the edsl #687 
- [x] Expose AddN operation to the edsl (required to aggregate tree predictions) #703
## Binary classification
- [x] In addition to the features required by regression task, we need to process the decision trees output to predict the class and probability (requiring `sigmoid` https://github.com/tf-encrypted/runtime/issues/641) associated with each class. See issue #639 for more detail.

## Multi-class classification
- [x] For muli-class classification, XGBoost uses the [ones-vs-rest](https://scikit-learn.org/stable/modules/generated/sklearn.multiclass.OneVsRestClassifier.html) approach where we train a model for each class. So if we train an XGBoost model limited to 1 tree ([n_estimators=1](https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBClassifier)), if we try to predict 3 different classes, we will end up 3 different decision trees (one for each class). So when aggregating the weights (as for Regression and binary classification) from each decision tree, we need to group/aggregate the decision trees by class. For multi-class classification we need to return predicted class and probability (might require `softmax` but to be validated) associated with each class. See #640.
